packages/core/src/credential/Credential.ts

```diff
@@ -332,6 +339,43 @@ export async function verifyPresentation(
   })
 }
 
+/**
+ * Queries the attestation record for a credential and matches their data. Fails if no attestation exists, if it is revoked, or if the attester is unknown.
+ *
+ * @param credential The [[ICredential]] whose attestation status should be checked.
+ * @param trustedAttesters A list of one or more attesters who are accepted as issuers of the credential. If the actual attester is not in trustedAttesters, and this is not a delegated attestation, verification fails.
+ * @param trustedDelegators An optional list of accepted delegators. If this is a delegated attestation and its issuer is not trusted, verification will still be successful if any of these identities is the owner of an ancestor node in the delegation tree.
+ */
+export async function verifyAttestation(
+  credential: ICredential,
+  trustedAttesters: DidUri[],
+  trustedDelegators: DidUri[] = []
+): Promise<void> {
+  const api = ConfigService.get('api')
+  const { rootHash } = credential
+  const maybeAttestation = await api.query.attestation.attestations(rootHash)
+  if (maybeAttestation.isNone)
+    throw new SDKErrors.CredentialUnverifiableError('Attestation not found')
+  const attestation = Attestation.fromChain(maybeAttestation, rootHash)
+  if (
+    credential.claim.cTypeHash !== attestation.cTypeHash ||
+    credential.delegationId !== attestation.delegationId
+  ) {
+    throw new SDKErrors.CredentialUnverifiableError(
+      'Attestation does not match credential'
+    )
+  }
+  if (attestation.revoked)
+    throw new SDKErrors.RevokedTypeError('Attestation revoked')
+  if (trustedAttesters.includes(attestation.owner)) return
```

Comment:
this should be an else. If we don't trust the attester, we would check if we trust the delegation root

Reply:
this is what's happening. no need for an else; it's an early return (return means success)

Reply:
Ah yes. Should have studied the code in more depth. Sorry.